### DISCUSSION TITLE
Nextjs mdx mantine components does not take theme from theme provider

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#2337] [Q&A]
I'm building an mdx based nextjs blog with Mantine.

I wrapped everything in _app.tsx inside `MantineProvider` and provided custom colors and changed primary color (to skyblue).

And in mdx files, I imported Mantine components, but the rendered components are still the default theme (blue).

Am I missing something here? Or is it just impossible?

**Comment by rtivital** [comment]
Mantine documentation works correctly with mdx, you view source code as reference – https://github.com/mantinedev/mantine/tree/master/docs

I haven't tried it with nextjs mdx, so cannot help with that setup.

**Comment by Toshinaki** [comment]
I figured it out!
It's not a problem of Nextjs or Mantine, it's a `mdx-bundler` config issue. (I'm using this package to manage mdx parsing)

I have to set the `globals` when using `bundleMDX()`: https://github.com/kentcdodds/mdx-bundler#globals

This is what I got so far:
```typescript
// process the mdx files
bundleMDX({
  ...
  globals: {
    "@mantine/core": {
      varName: "globalMantine",
      namedExports: ["Button"],
      defaultExport: false,
    },
  },
});

// in [slug].tsx
import * as Mantine from "@mantine/core";
...
const Component = useMemo(
  () => getMDXComponent(post.content, { globalMantine: Mantine }),
  [post.content]
);
...
```
Then the components inside mdx files can use the same Mantine with page components, and the theme is also applied.

Though I have to name all imports I need by setting `namedExports: ["Button"]`. And this list will grow when used components increase.
Hope someone can share a better solution.
